### PR TITLE
Setup and apply fourmolu code formatter

### PR DESCRIPTION
**Pull request by koslambrou**
See the commit 86750f67fe7ce96ad867dc2d728b7834dcf499fc for the whole fourmolu setup in the Nix shell. We removed stylish-haskell as well.

See the commit 95b0e96e94ad1a0dfe2e2834a4d908c5194b2684 for the actual application of fourmolu to the Marconi codebase.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting main unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [ ] Reviewer requested

### CODE REVIEW

File: marconi-chain-index/json-rpc/src/Network/JsonRpc/Client/Types.hs [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
+  (RunClient m, KnownSymbol method, ToJSON p, FromJSON e, FromJSON r)
+  => HasClient m (JsonRpc method p e r)
+  where
+  type
+    Client m (JsonRpc method p e r) =
+      p -> m (JsonRpcResponse e r)
+
+  clientWithRoute _ _ req p =
+    client req jsonRpcRequest
+   where
```
Comment (berewt) [inline marconi-chain-index/json-rpc/src/Network/JsonRpc/Client/Types.hs:47]: This `where` indentation looks weird to me. Don't we want `where` to be 2-indented and `where`-functions 4-indented?

Comment (koslambrou) [reply marconi-chain-index/json-rpc/src/Network/JsonRpc/Client/Types.hs]: There's an option for this. Here's what it looks like: c1921d478b5752ad016b6de0bf2dbdc91866e15c

**Review by berewt — APPROVED** [review]
Yes! Just a minor thing: can you also update the `.editorconfig` file to provide the same indentation and column limit for `*.hs` files?

**Review by kayvank — APPROVED** [review]

**Comment by eyeinsky** [comment]
Would it be possible to omit the files currently in ongoing PRs from this, as otherwise there will be merge conflicts? (The PRs could apply fourmolu within themselves.)

**Pull request merged**